File: Modules/WillStrohl.Injection/Entities/InjectionInfo.vb
'
' Will Strohl (will.strohl@gmail.com)
' http://www.willstrohl.com
'
'Copyright (c) 2009-2015, Will Strohl
'All rights reserved.
'
'Redistribution and use in source and binary forms, with or without modification, are 
'permitted provided that the following conditions are met:
'
'Redistributions of source code must retain the above copyright notice, this list of 
'conditions and the following disclaimer.
'
'Redistributions in binary form must reproduce the above copyright notice, this list 
'of conditions and the following disclaimer in the documentation and/or other 
'materials provided with the distribution.
'
'Neither the name of Will Strohl, Content Injection, nor the names of its contributors may be 
'used to endorse or promote products derived from this software without specific prior 
'written permission.
'
'THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY 
'EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES 
'OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT 
'SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
'INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED 
'TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR 
'BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
'CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
'ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH 
'DAMAGE.
'

Imports DotNetNuke.Common.Utilities
Imports DotNetNuke.Entities.Modules
Imports DotNetNuke.Services.Exceptions

Imports WillStrohl.Modules.Injection.InjectionInfoMembers

Namespace WillStrohl.Modules.Injection

    <Serializable()> _
    Public NotInheritable Class InjectionInfo
        Implements IInjectionInfo, IHydratable

#Region " Private Members "

        Private p_InjectName As String = String.Empty
        Private p_InjectContent As String = String.Empty
        Private p_InjectionId As Integer = Null.NullInteger
        Private p_InjectTop As Boolean = Null.NullBoolean
        Private p_IsEnabled As Boolean = Null.NullBoolean
        Private p_ModuleId As Integer = Null.NullInteger
        Private p_OrderShown As Integer = Null.NullInteger

#End Region

#Region " Public Properties "

        Public Property InjectName() As String Implements IInjectionInfo.InjectName
            Get
                Return Me.p_InjectName
            End Get
            Set(ByVal value As String)
                Me.p_InjectName = value
            End Set
        End Property

        Public Property InjectContent() As String Implements IInjectionInfo.InjectContent
            Get
                Return Me.p_InjectContent
            End Get
            Set(ByVal value As String)
                Me.p_InjectContent = value
            End Set
        End Property

        Public Property InjectionId() As Integer Implements IInjectionInfo.InjectionId
            Get
                Return Me.p_InjectionId
            End Get
            Set(ByVal value As Integer)
                Me.p_InjectionId = value
            End Set
        End Property

        Public Property InjectTop() As Boolean Implements IInjectionInfo.InjectTop
            Get
                Return Me.p_InjectTop
            End Get
            Set(ByVal value As Boolean)
                Me.p_InjectTop = value
            End Set
        End Property

        Public Property IsEnabled() As Boolean Implements IInjectionInfo.IsEnabled
            Get
                Return Me.p_IsEnabled
            End Get
            Set(ByVal value As Boolean)
                Me.p_IsEnabled = value
            End Set
        End Property

        Public Property ModuleId() As Integer Implements IInjectionInfo.ModuleId
            Get
                Return Me.p_ModuleId
            End Get
            Set(ByVal value As Integer)
                Me.p_ModuleId = value
            End Set
        End Property

        Public Property OrderShown() As Integer Implements IInjectionInfo.OrderShown
            Get
                Return Me.p_OrderShown
            End Get
            Set(ByVal value As Integer)
                Me.p_OrderShown = value
            End Set
        End Property

#End Region

        Public Sub New()
            ' do nothing
        End Sub

#Region " IHydratable Implementation "

        Public Property KeyID() As Integer Implements DotNetNuke.Entities.Modules.IHydratable.KeyID
            Get
                Return Me.p_InjectionId
            End Get
            Set(ByVal value As Integer)
                Me.p_InjectionId = value
            End Set
        End Property

        Public Sub Fill(ByVal dr As System.Data.IDataReader) Implements DotNetNuke.Entities.Modules.IHydratable.Fill
            Try
                While dr.Read
                    If Not dr.Item(InjectionIdField) Is Nothing Then
                        Me.p_InjectionId = Integer.Parse(dr.Item(InjectionIdField).ToString, Globalization.NumberStyles.Integer)
                    End If
                    If Not dr.Item(InjectNameField) Is Nothing Then
                        Me.p_InjectName = dr.Item(InjectNameField).ToString
                    End If
                    If Not dr.Item(InjectContentField) Is Nothing Then
                        Me.p_InjectContent = dr.Item(InjectContentField).ToString
                    End If
                    If Not dr.Item(InjectTopField) Is Nothing Then
                        Me.p_InjectTop = Boolean.Parse(dr.Item(InjectTopField).ToString)
                    End If
                    If Not dr.Item(IsEnabledField) Is Nothing Then
                        Me.p_IsEnabled = Boolean.Parse(dr.Item(IsEnabledField).ToString)
                    End If
                    If Not dr.Item(ModuleIdField) Is Nothing Then
                        Me.p_ModuleId = Integer.Parse(dr.Item(ModuleIdField).ToString, Globalization.NumberStyles.Integer)
                    End If
                    If Not dr.Item(OrderShownField) Is Nothing Then
                        Me.p_OrderShown = Integer.Parse(dr.Item(OrderShownField).ToString, Globalization.NumberStyles.Integer)
                    End If
                End While
            Catch ex As Exception
                LogException(ex)
            Finally
                If Not dr.IsClosed Then
                    dr.Close()
                End If
            End Try
        End Sub

#End Region

    End Class

End Namespace

File: Modules/WillStrohl.Injection/Entities/InjectionInfo.cs
/*
Copyright © Upendo Ventures, LLC

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
associated documentation files (the "Software"), to deal in the Software without restriction, 
including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, 
and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, 
subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial 
portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT 
NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND 
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES 
OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

using DotNetNuke.Common.Utilities;
using DotNetNuke.Services.Exceptions;
using System;
using System.Collections.Generic;
using System.Data;
using System.Globalization;
using DotNetNuke.Entities.Modules;

namespace WillStrohl.Modules.Injection.Entities
{
    [Serializable]
    public sealed class InjectionInfo : IInjectionInfo, IHydratable
    {
        #region Private Members

        private string p_InjectName = string.Empty;
        private string p_InjectContent = string.Empty;
        private int p_InjectionId = Null.NullInteger;
        private bool p_InjectTop = Null.NullBoolean;
        private bool p_IsEnabled = Null.NullBoolean;
        private int p_ModuleId = Null.NullInteger;
        private int p_OrderShown = Null.NullInteger;
        private List<CustomPropertyInfo> p_CustomProperties = null;

        #endregion

        #region Public Properties

        public string InjectName
        {
            get { return p_InjectName; }
            set { p_InjectName = value; }
        }

        public string InjectContent
        {
            get { return p_InjectContent; }
            set { p_InjectContent = value; }
        }

        public int InjectionId
        {
            get { return p_InjectionId; }
            set { p_InjectionId = value; }
        }

        public bool InjectTop
        {
            get { return p_InjectTop; }
            set { p_InjectTop = value; }
        }

        public bool IsEnabled
        {
            get { return p_IsEnabled; }
            set { p_IsEnabled = value; }
        }

        public int ModuleId
        {
            get { return p_ModuleId; }
            set { p_ModuleId = value; }
        }

        public int OrderShown
        {
            get { return p_OrderShown; }
            set { p_OrderShown = value; }
        }

        public List<CustomPropertyInfo> CustomProperties
        {
            get { return p_CustomProperties; }
            set { p_CustomProperties = value; }
        }

        #endregion

        public InjectionInfo()
        {
            p_CustomProperties = new List<CustomPropertyInfo>();
        }

        #region IHydratable Implementation

        public int KeyID
        {
            get { return p_InjectionId; }
            set { p_InjectionId = value; }
        }

        public void Fill(IDataReader dr)
        {
            try
            {
                while (dr.Read())
                {
                    if (dr[InjectionInfoMembers.InjectionIdField] != null)
                    {
                        p_InjectionId = int.Parse(dr[InjectionInfoMembers.InjectionIdField].ToString(), NumberStyles.Integer);
                    }

                    if (dr[InjectionInfoMembers.InjectNameField] != null)
                    {
                        p_InjectName = dr[InjectionInfoMembers.InjectNameField].ToString();
                    }

                    if (dr[InjectionInfoMembers.InjectContentField] != null)
                    {
                        p_InjectContent = dr[InjectionInfoMembers.InjectContentField].ToString();
                    }

                    if (dr[InjectionInfoMembers.InjectTopField] != null)
                    {
                        p_InjectTop = bool.Parse(dr[InjectionInfoMembers.InjectTopField].ToString());
                    }

                    if (dr[InjectionInfoMembers.IsEnabledField] != null)
                    {
                        p_IsEnabled = bool.Parse(dr[InjectionInfoMembers.IsEnabledField].ToString());
                    }

                    if (dr[InjectionInfoMembers.ModuleIdField] != null)
                    {
                        p_ModuleId = int.Parse(dr[InjectionInfoMembers.ModuleIdField].ToString(), NumberStyles.Integer);
                    }

                    if (dr[InjectionInfoMembers.OrderShownField] != null)
                    {
                        p_OrderShown = int.Parse(dr[InjectionInfoMembers.OrderShownField].ToString(), NumberStyles.Integer);
                    }

                    try
                    {
                        if (dr[InjectionInfoMembers.CustomPropertiesField] != null)
                        {
                            var properties = dr[InjectionInfoMembers.CustomPropertiesField].ToString();
                            p_CustomProperties = properties.FromJson<List<CustomPropertyInfo>>();
                        }
                    }
                    catch
                    {
                    }
                }
            }
            catch (Exception ex)
            {
                Exceptions.LogException(ex);
            }
            finally
            {
                if (!dr.IsClosed)
                {
                    dr.Close();
                }
            }
        }

        #endregion
    }
}